src/include/cursor.h

```diff
@@ -117,7 +117,8 @@ struct __wt_cursor_btree {
     /* Next item(s) found during search */
     WT_INSERT *next_stack[WT_SKIP_MAXDEPTH];
 
-    uint32_t page_deleted_count; /* Deleted items on the page */
+    uint32_t page_obsolete_deleted_count; /* Obsolete deleted items on the page */
+    bool non_deleted_updates;             /* Any valid items on the page */
```

Comment:
I was wondering if we could make this a flag?

Reply:
I got your point of adding it as a cursor btree flag value instead of a variable in the structure. I am just thinking which approach has more readability.

Reply:
I changed into a flag instead of a variable.